src/formatDate.js

```diff
@@ -7,8 +7,35 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatedDate = [];
+  const splitedDate = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    for (let j = 0; j < toFormat.length - 1; j++) {
+      if (fromFormat[i] === 'YYYY' && toFormat[j] === 'YY') {
+        formatedDate[j] = splitedDate[i].slice(2);
+        break;
+      }
+
+      if (fromFormat[i] === 'YY' && toFormat[j] === 'YYYY') {
+        if (splitedDate[i] < 30) {
+          formatedDate[j] = '20' + splitedDate[i];
+        } else {
+          formatedDate[j] = '19' + splitedDate[i];
+        }
+
+        break;
+      }
+
+      if (fromFormat[i] === toFormat[j]) {
+        formatedDate[j] = splitedDate[i];
+      }
+    }
```

Comment:
The nested loop approach is inefficient and may not work correctly if the order of the date parts in 'toFormat' is different from 'fromFormat'. Instead, you should map the indices of the date parts in 'fromFormat' to their corresponding values in 'splitedDate', and then build the formatted date based on the 'toFormat' array.